cli/azd/pkg/account/manager.go

```diff
@@ -62,27 +62,27 @@ func NewManager(configManager config.Manager, azCli azcli.AzCli) (*Manager, erro
 // Gets the default subscription for the logged in account.
 // 1. Returns AZD config defaults if exists
 // 2. Returns AZ CLI defaults if exists
-func (m *Manager) GetAccountDefaults(ctx context.Context) *Account {
+func (m *Manager) GetAccountDefaults(ctx context.Context) (*Account, error) {
 	subscription, err := m.getDefaultSubscription(ctx)
-
-	// If we don't have a default subscription then the principal does not have any active
-	// subscriptions or the configured value is not valid.
 	if err != nil {
-		subscription = nil
+		return nil, fmt.Errorf("failed retrieving default subscription: %w", err)
 	}
 
-	location, err := m.getDefaultLocation(ctx, subscription.Id)
+	var location *Location
 
-	// If we don't have a default location then either this is the first run experience
-	// or the location specified in configuration is invalid.
-	if err != nil {
+	if subscription == nil {
 		location = &defaultLocation
```

Comment:
Can the user set a config.default location without setting a config.default subscription??

If yes, are we ignoring the config.default location set by the user here?  And always using `defaultLocation` whenever there's not a subscription default config?

Reply:
The locations are dependent on subscription.  The only ARM locations API always expects a subscription scope.
https://learn.microsoft.com/en-us/rest/api/resources/subscriptions/list-locations?tabs=HTTP

Reply:
The user can set it manually in the config, but not through any of our flows.

Reply:
> The user can set it manually in the config, but not through any of our flows.

When the user sets it manually, this line is ignoring that the value that the user set in the config. (when the subscription is not set).  We need to block users from setting the location manually without having a subscription.

Reply:
I'm not going to add any special cases into `azd config set` at this time.